### PR TITLE
WEBC-636

### PR DESCRIPTION
**Pull request by igorshevach**

### CODE REVIEW

File: Player/IMediaElement.cs [hunk 1]
```diff
@@ -54,6 +57,8 @@ public interface IMediaElement
 
          event  EventHandler<MouseButtonEventArgs> MouseLeftButtonUp;
 
+         event EventHandler<DateTimeArgs> SyncPointPlayed;
```
Comment (kalturaguy) [inline Player/IMediaElement.cs]: this is not needed, SendEvent is generic one for all events

File: Player/MainPage.xaml.cs [hunk 2]
```diff
@@ -395,6 +408,11 @@ void media_MediaEnded(object sender, RoutedEventArgs e)
             SendEvent("playerPlayEnd");
         }
 
+        void media_SyncPointPlayed(object sender, DateTimeArgs e)
+        {
+            SendEvent("syncPointPlayed","{\"syncpoint\":\"" + e.CurrentTime.ToString() + "\"}");
```
Comment (kalturaguy) [inline Player/MainPage.xaml.cs]: i think the syncpoint should contain unix time and not string (harder to parse)

File: Player/IMediaElement.cs (hunk 1 above)
Comment (igorshevach) [inline Player/IMediaElement.cs]: SendEvent is called from withing SyncPointPlayed handler and is not visible by imediaelement implementor. how the event will be raised otherwised?

File: Player/MainPage.xaml.cs (hunk 2 above)
Comment (igorshevach) [inline Player/MainPage.xaml.cs]: unix time expressed as seconds , milliseconds, fraction?

Comment (igorshevach) [inline Player/MainPage.xaml.cs]: changed to unix time

File: Player/IMediaElement.cs (hunk 1 above)
Comment (igorshevach) [inline Player/IMediaElement.cs]: need this event to deliver unix timestamp upstack

File: Player/MainPage.xaml.cs [hunk 3]
```diff
@@ -160,7 +160,7 @@ private void InitMedia()
         private void InitTimer()
         {
             if (_timerRate == 0)
-                _timerRate = 250;
+                _timerRate = 100;
```
Comment (kalturaguy) [inline Player/MainPage.xaml.cs]: keep the 250ms resolution

File: Player/IMediaElement.cs [hunk 4]
```diff
@@ -1,4 +1,5 @@
-﻿using System;
+﻿using MediaStreamSrc.Classes;
```
Comment (kalturaguy) [inline Player/IMediaElement.cs]: this file hasn't changed, can you remove the first line

File: Player/Player.csproj [hunk 5]
```diff
@@ -62,7 +63,7 @@
   <ItemGroup>
     <Reference Include="Microsoft.Web.Media.SmoothStreaming, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\..\..\..\Program Files (x86)\Microsoft SDKs\IIS Smooth Streaming Client\v2.0\Silverlight\Microsoft.Web.Media.SmoothStreaming.dll</HintPath>
+      <HintPath>..\..\Microsoft.Web.Media.SmoothStreaming.dll</HintPath>
```
Comment (kalturaguy) [inline Player/Player.csproj]: ?

File: .gitignore [hunk 6]
```diff
@@ -107,3 +107,6 @@ _UpgradeReport_Files/
 Backup*/
 UpgradeLog*.XML
 Player.Web/ClientBin/Player.xap
+Player/multicast-silverlight-lib.dll
```
Comment (kalturaguy) [inline .gitignore]: this should exist Player/multicast-silverlight-lib.dll

**Pull request merged**